src/common/systems/GameController.hpp

```diff
@@ -50,8 +53,9 @@ class GameController {
     sf::Clock m_clock;
     sf::Int32 m_elapsedTime = 0;
     sf::Int32 m_timeAccumulator = 0;
-    sf::Int32 m_stepCount = 0;
-    sf::Int32 m_updateCount = 0;
+
+    TemporalMetric m_stepMetric{120, sf::seconds(0.5f)};
+    TemporalMetric m_updateMetric{120, sf::seconds(0.5f)};
```

Comment:
rename these to be more specific
maybe `m_gameStepMetric` and `m_gameUpdateMetric `